x/tokenfactory/keeper/admins_test.go

```diff
@@ -1,74 +1,262 @@
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestChangeAdminDenom() {
+	var (
+		denom    string
+		addr0bal int64
+		addr1bal int64
+	)
+
+	for _, tc := range []struct {
+		desc  string
+		setup func()
+		valid bool
+	}{
+		{
+			desc: "creator admin can't mint after setting to '' ",
+			setup: func() {
+				_, err := suite.msgServer.ChangeAdmin(sdk.WrapSDKContext(suite.Ctx), types.NewMsgChangeAdmin(suite.TestAccs[0].String(), denom, ""))
+				queryRes, err := suite.queryClient.DenomAuthorityMetadata(suite.Ctx.Context(), &types.QueryDenomAuthorityMetadataRequest{
+					Denom: denom,
+				})
+				suite.Require().NoError(err)
+				suite.Require().Equal("", queryRes.AuthorityMetadata.Admin)
+
+				// can't mint.
+				_, err = suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(suite.TestAccs[0].String(), sdk.NewInt64Coin(denom, 5)))
+				suite.Require().Error(err)
+				suite.Require().Equal(suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], denom).Amount.Int64(), addr0bal)
+			},
+		},
+		{
+			desc: "non-admins can't change the existing admin",
+			setup: func() {
+				// Test Change Admin
+				_, err := suite.msgServer.ChangeAdmin(sdk.WrapSDKContext(suite.Ctx), types.NewMsgChangeAdmin(suite.TestAccs[1].String(), denom, suite.TestAccs[2].String()))
+				queryRes, err := suite.queryClient.DenomAuthorityMetadata(suite.Ctx.Context(), &types.QueryDenomAuthorityMetadataRequest{
+					Denom: denom,
+				})
+				suite.Require().NoError(err)
+				suite.Require().NotEqual(suite.TestAccs[2].String(), queryRes.AuthorityMetadata.Admin)
+			},
+		},
+		{
+			desc: "success change admin",
+			setup: func() {
+				// Test Change Admin
+				_, err := suite.msgServer.ChangeAdmin(sdk.WrapSDKContext(suite.Ctx), types.NewMsgChangeAdmin(suite.TestAccs[0].String(), denom, suite.TestAccs[1].String()))
+				queryRes, err := suite.queryClient.DenomAuthorityMetadata(suite.Ctx.Context(), &types.QueryDenomAuthorityMetadataRequest{
+					Denom: denom,
+				})
+				suite.Require().NoError(err)
+				suite.Require().Equal(suite.TestAccs[1].String(), queryRes.AuthorityMetadata.Admin)
+
+				// New admin can mint.
+				_, err = suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(suite.TestAccs[1].String(), sdk.NewInt64Coin(denom, 5)))
+
+				addr1bal += 5
+				suite.Require().NoError(err)
+				suite.Require().Equal(suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], denom).Amount.Int64(), addr0bal)
+			},
+		},
+	} {
+		suite.Run(fmt.Sprintf("Case %s", tc.desc), func() {
+			// setup test
+			suite.SetupTest()
+			addr0bal = 0
+			addr1bal = 0
+
+			// Create a denom and mint
+			res, err := suite.msgServer.CreateDenom(sdk.WrapSDKContext(suite.Ctx), types.NewMsgCreateDenom(suite.TestAccs[0].String(), "bitcoin"))
```

Comment:
same as comment left in `keeper_test.go` we can either utilzie that better or we can replace that by directly calling the methods from the test, i would prefer the latter option